### PR TITLE
Xref name property

### PR DESCRIPTION
**Pull request by mingwli**
[AB#192954](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/192954)

[migration-tool pr](https://apidrop.visualstudio.com/Content%20CI/_git/docfx-v3-migration/pullrequest/843?_a=overview)
###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5668)

### CODE REVIEW

File: src/docfx/build/xref/InternalXrefSpec.cs [hunk 1]
```diff
@@ -9,8 +9,12 @@ namespace Microsoft.Docs.Build
 {
     internal class InternalXrefSpec : IXrefSpec
     {
+        private readonly Lazy<string?> _name;
+
         public SourceInfo<string> Uid { get; }
 
+        public string Name => _name.Value ?? Uid;
```
Comment (yufeih) [inline src/docfx/build/xref/InternalXrefSpec.cs]:
```suggestion
        public string Name => GetXrefPropertyValueAsString("name") ?? Uid;
```

File: src/docfx/build/xref/ExternalXrefSpec.cs [hunk 2]
```diff
@@ -4,15 +4,24 @@
 using System.Collections.Generic;
 using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
+using YamlDotNet.Core.Tokens;
```
Comment (yufeih) [inline src/docfx/build/xref/ExternalXrefSpec.cs]: remove?

File: src/docfx/build/xref/InternalXrefMapBuilder.cs [hunk 3]
```diff
@@ -93,7 +93,7 @@ private static (Error? error, InternalXrefSpec? spec) LoadMarkdown(Context conte
                 return default;
             }
 
-            var xref = new InternalXrefSpec(metadata.Uid, file.SiteUrl, file);
+            var xref = new InternalXrefSpec(metadata.Uid, file.SiteUrl, file); var name = new Lazy<string?>(() => string.IsNullOrEmpty(metadata.Title) ? metadata.Uid : metadata.Title);
```
Comment (yufeih) [inline src/docfx/build/xref/InternalXrefMapBuilder.cs]: revert?

File: src/docfx/build/xref/ExternalXrefSpec.cs [hunk 4]
```diff
@@ -9,10 +9,18 @@ namespace Microsoft.Docs.Build
 {
     internal class ExternalXrefSpec : IXrefSpec
     {
+        private string? _name;
+
         public string Uid { get; set; } = "";
 
         public string Href { get; set; } = "";
 
+        public string Name
```
Comment (live1206) [inline src/docfx/build/xref/ExternalXrefSpec.cs]: nit: does this equal to `Name { get; set; } = Uid`?

**Comment by mingwli** [comment]
> ⚠[DevSandbox](https://github.com/OPS-E2E-PPE/DevSandbox)(00:00:16.3748339, 0 more diff)

expected name fallback to uid and export

**Comment by mingwli** [comment]
> ⚠[azure-docs-pr](https://github.com/OPS-E2E-PPE/azure-docs-pr)(00:01:57.7959050, 16706 more diff)

CRR update

**Pull request merged**